run/filesystem/e2e_test.py

```diff
@@ -0,0 +1,135 @@
+import os
+import subprocess
+from urllib import request
+import uuid
+
+import pytest
+
+# Unique suffix to create distinct service names
+SUFFIX = uuid.uuid4().hex[:10]
+PROJECT = os.environ['GOOGLE_CLOUD_PROJECT']
+
+
+@pytest.fixture
+def deployed_service():
+    # Deploy image to Cloud Run
+    service_name = f'filesystem-{SUFFIX}'
+    connector = os.environ['CONNECTOR']
+    ip_address = os.environ['IP_ADDRESS']
+
+    subprocess.run(
+        [
+            'gcloud',
+            'builds',
+            'submit',
+            '--config',
+            'cloudbuild.yaml',
+            '--project',
+            PROJECT,
+            f'--substitutions=_SERVICE_NAME={service_name},_IP_ADDRESS={ip_address},_CONNECTOR={connector}'
+        ],
+        check=True,
+    )
+
+    yield service_name
+
+    subprocess.run(
+        [
+            'gcloud',
+            'run',
+            'services',
+            'delete',
+            service_name,
+            '--region=us-central1',
+            '--platform=managed',
+            '--quiet',
+            '--project',
+            PROJECT,
+        ],
+        check=True,
+    )
+
+
+@pytest.fixture
+def service_url_auth_token(deployed_service):
+    # Get Cloud Run service URL and auth token
+    service_url = (
+        subprocess.run(
+            [
+                'gcloud',
+                'run',
+                'services',
+                'describe',
+                deployed_service,
+                '--region=us-central1',
+                '--platform=managed',
+                '--format=value(status.url)',
+                '--project',
+                PROJECT,
+            ],
+            stdout=subprocess.PIPE,
+            check=True,
+        )
+        .stdout.strip()
+        .decode()
+    )
+    auth_token = (
+        subprocess.run(
+            ['gcloud', 'auth', 'print-identity-token'],
+            stdout=subprocess.PIPE,
+            check=True,
+        )
+        .stdout.strip()
+        .decode()
+    )
+
+    yield service_url, auth_token
+
+    # no deletion needed
+
+
+def test_end_to_end(service_url_auth_token):
+    service_url, auth_token = service_url_auth_token
+    # Non mnt directory
+    req = request.Request(
+        service_url, headers={'Authorization': f'Bearer {auth_token}'}
+    )
+    response = request.urlopen(req)
+    assert response.status == 200
+
+    # Mnt directory
+    mnt_url = f'{service_url}/mnt/nfs/filestore'
+    req = request.Request(
+        mnt_url, headers={'Authorization': f'Bearer {auth_token}'}
+    )
+    response = request.urlopen(req)
+    assert response.status == 200
+
+    date = datetime.datetime.utcnow()
+    body = response.read()
+    weekday = date.strftime('%a')
+    month = date.strftime('%b')
+    day = date.strftime('%d')
+    assert f'{weekday}-{month}-{day}' in body.decode()  # Date
+    hour = date.strftime('%H')
+    minute = date.strftime('%M')
+    assert f'{hour}:{minute}' in body.decode()  # Time
```

Comment:
If the service and client machines times are not in sync, or even if the clock ticks from one minute to the next between the server write and client test, this can fail. It's not likely, but if it's a concern, perhaps a nonce in the request and have that written into the file, too.